### DISCUSSION TITLE
TS fontWeight error

### DISCUSSION THREAD

**Discussion by HellYeahOmg** [jpudysz/react-native-unistyles#57] [Q&A] [answered]
I was trying to follow the instructions on creating variants for components and i got fontWeight ts error (no changes were made to styles from the https://reactnativeunistyles.vercel.app/example/variants/ BaseText example)
<img width="946" alt="image" src="https://github.com/jpudysz/react-native-unistyles/assets/29250449/0e52c07f-bc9c-49c0-8fc3-3824a3724df0">

But, if i manually set up fontWeight in createStyleSheet everything seems to be fine
`const stylesheet = createStyleSheet((theme) => ({
  body: {
    ...theme.components.typography.body,
    fontWeight: '400',
  },
  bodyBold: {
    ...theme.components.typography.bodyBold,
    fontWeight: '400',
  },
}))`

**Comment by ridvanaltun** [comment]
I think you need to create a bug report for it.

**Comment by jpudysz** [comment, marked as answer]
I received the same report twice today! 😅 
That's an issue with React Native types - please check the comments here:

https://github.com/jpudysz/react-native-unistyles/issues/51

I will add this information to the documentation tomorrow. 🙏

**Comment by HellYeahOmg** [comment]
Hm, I don't think "as const" fixed it for me.

My theme:
`export const theme = {
  colors: {
     ...
  },
  components: {
    typography: {
      h1: {
        fontFamily: 'Mulish',
        fontSize: 42,
        fontWeight: 700,
        lineHeight: 52,
      },
      // other texts with same properties, just different values
    },
  },
} as const`

Keep getting the same error
<img width="944" alt="Screenshot 2023-11-04 at 23 56 35" src="https://github.com/jpudysz/react-native-unistyles/assets/29250449/b51f00ad-4ea3-433f-8bed-61ee46ec1932">

<img width="993" alt="Screenshot 2023-11-04 at 23 58 36" src="https://github.com/jpudysz/react-native-unistyles/assets/29250449/4fe34c0f-5d9c-479a-b38f-492629dfd93a">

**Comment by jpudysz** [comment]
It's because you have invalid `fontWeight`. Font weights are strings not numbers. 
Error says - Type '400' (number) is not assignable to "bold", "normal", "100" .. etc
